library/pod-security-policy/fsgroup/template.yaml

```diff
@@ -44,62 +44,76 @@ spec:
                     type: integer
   targets:
     - target: admission.k8s.gatekeeper.sh
-      rego: |
-        package k8spspfsgroup
+      code:
+      - engine: K8sNativeValidation
+        source:
+          variables:
+          - name: fsGroup
+            expression: '!has(variables.anyObject.spec.securityContext) ? "" : !has(variables.anyObject.spec.securityContext.fsGroup) ? "" : variables.anyObject.spec.securityContext.fsGroup'
+          - name: input_fsGroup_allowed
+            expression: |
+              !has(variables.params.rule) ? true : variables.params.rule == "RunAsAny" ? true : variables.params.rule == "MayRunAs" && variables.fsGroup == "" ? true : (variables.params.rule == "MayRunAs" || variables.params.rule == "MustRunAs") && has(variables.params.ranges) && size(variables.params.ranges) > 0 ? variables.params.ranges.all(range, range.min <= variables.fsGroup && range.max >= variables.fsGroup) : false
+          validations:
+          - expression: '(has(request.operation) && request.operation == "UPDATE") || variables.input_fsGroup_allowed'
+            messageExpression: '"The provided pod spec fsGroup is not allowed, pod: " + variables.anyObject.metadata.name + ". Allowed fsGroup: " + variables.params.rule'
```

Comment:
The error string will end with something like "Allowed fsGroup: MustRunAs" which I don't think is what you intended.

Reply:
New message:
> The provided pod spec fsGroup is not allowed, pod: fsgroup-disallowed. Allowed fsGroup rule: MustRunAs, allowed fsGroup ranges: empty